### PR TITLE
Remove NotImplementedError for parse_dates keyword in read_excel

### LINKED ISSUE #11544: ERR: raise NotImplemented error if keywords are passed to read_excel which are not supported
xref #11527 
xref #10001

currently we pass thru keywords to the `TextReader` but some of these don't do anything. Should catch and raise on these.

### PR DESCRIPTION
**Pull request by jreback**
rebase of #12051 and fixes on top

closes #11544

### CODE REVIEW

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -1176,13 +1176,18 @@ def _should_parse_dates(self, i):
         if isinstance(self.parse_dates, bool):
             return self.parse_dates
         else:
-            name = self.index_names[i]
+            if self.index_names is not None:
+                name = self.index_names[i]
```
Comment (jreback) [inline pandas/io/parsers.py]: @jorisvandenbossche this fixes, though not quite sure why this could/would be ``None`` here.

@gfyoung any idea?

**Comment by jreback** [comment]
I'll fix the authorship on merge.

**Comment by jorisvandenbossche** [comment]
Thanks for resurrecting this! Not directly an idea about your question (that was the reason I didn't update the PR, as I didn't found the time to dive into the parsers code to figure this out)

File: pandas/io/parsers.py (hunk 1 above)
Comment (gfyoung) [inline pandas/io/parsers.py]: `index_names` is `None` by default, though why that wasn't caught before is strange.

Comment (jreback) [inline pandas/io/parsers.py]: right. yep must not have taken that path at all, e.g. ``read_csv(..., parse_dates=['column_name'], index_col=0)`` is all that this is doing.......

if you want to look and see maybe missing something obvious, would be great.

File: pandas/io/excel.py [hunk 2]
```diff
@@ -343,13 +343,10 @@ def _parse_excel(self, sheetname=0, header=0, skiprows=None, names=None,
...
-        if parse_dates:
-            raise NotImplementedError("parse_dates keyword of read_excel "
-                                      "is not implemented")
 
-        if date_parser is not None:
-            raise NotImplementedError("date_parser keyword of read_excel "
-                                      "is not implemented")
+        if parse_dates is True and index_col is None:
+            warn("The 'parse_dates=True' keyword of read_excel was provided"
+                 " without an 'index_col' keyword value.")
```
Comment (gfyoung) [inline pandas/io/excel.py]: Can we add a test that hits this warning?

File: pandas/io/parsers.py (hunk 1 above)
Comment (gfyoung) [inline pandas/io/parsers.py]: Sure thing.  Just curious, what test was failing beforehand that allowed you to catch this?

Comment (jreback) [inline pandas/io/parsers.py]: @jorisvandenbossche re-enabled the ``parse_dates`` kw in ``read_excel``, but was failing on this. Note that this by-definition is *only* python engine. (as that is what excel uses).

File: pandas/io/excel.py (hunk 2 above)
Comment (jreback) [inline pandas/io/excel.py]: added, though I am puzzled why a simple ``parse_dates=True`` doesn't just work.....

**Comment by jreback** [comment]
@jorisvandenbossche good with this (it all passed ). ?

**Pull request closed without merging**